apps/ca.c

```diff
@@ -246,7 +246,8 @@ int ca_main(int argc, char **argv)
     const char *serialfile = NULL, *subj = NULL;
     char *prog, *startdate = NULL, *enddate = NULL;
     char *dbfile = NULL, *f, *randfile = NULL;
-    char buf[3][BSIZE];
+    char new_cert[BSIZE] = { 0 };
```

Comment:
interesting, might be better to put put ```new_cert[0] == '\0';``` at the start of the code, because this requires the whole certs be initialized.  maybe.